pkg/grpc/grpc.go

```diff
@@ -58,25 +60,43 @@ func NewGRPCClientFromConfiguration(configuration *configuration.ClientConfigura
 		dialOptions = append(dialOptions, grpc.WithInsecure())
 	}
 
+	if oauthConfig := config.Oauth; oauthConfig != nil {
+		var perRPC credentials.PerRPCCredentials
+		var err error
+		switch oauthConfig.Credentials.(type) {
```

Comment:
If you use `switch credentials := oauthConfig.Credentials.(type)` here, you can use `credentials.ServiceAccountKey` later on to access it without any type assertions.

Reply:
done